File: python/DUE/Pin.py
import time

class PinController:   
     
    def __init__(self):
        pass


    def get_button_a(self):
        return 97

    def get_button_b(self):
        return 98
    
    def get_led(self):
        return 108

    def get_piezo(self):
        return 112
    
    def get_pullnone(self):
        return 0
    
    def get_pullup(self):
        return 1
    
    def get_pulldown(self):
        return 2

    def set_empty(self, value: int):
        return    
    
    ButtonA = property(get_button_a, set_empty)  
    ButtonB = property(get_button_b, set_empty)   
    Led = property(get_led, set_empty) 
    Piezo = property(get_piezo, set_empty)
    PullNone = property(get_pullnone, set_empty)
    PullUp = property(get_pullup, set_empty)
    PullDown = property(get_pulldown, set_empty)

File: dotnet/GHIElectronics.DUELink/Pin.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GHIElectronics.DUELink {
    public partial class DUELinkController {
        public class PinController {

            internal static int BUTTON_A = 97;
            internal static int BUTTON_B = 98;
            internal static int LED = 108;
            internal static int PIEZO = 112;
            internal static int PULLNONE = 0;
            internal static int PULLUP = 1;
            internal static int PULLDOWN = 2;

            internal static int BUTTON_U = 85;
            internal static int BUTTON_D = 68;
            internal static int BUTTON_L = 76;
            internal static int BUTTON_R = 82;


            public int ButtonA { get; } = BUTTON_A;
            public int ButtonB { get; } = BUTTON_B;
            public int ButtonUp { get; } = BUTTON_U;
            public int ButtonDown { get; } = BUTTON_D;
            public int ButtonLeft { get; } = BUTTON_L;
            public int ButtonRight { get; } = BUTTON_R;
            public int Led { get; } = LED;
            public int Piezo { get; } = PIEZO;
            public int PullNone { get; } = PULLNONE;
            public int PullUp { get; } = PULLUP;
            public int PullDown { get; } = PULLDOWN;

        }
    }
}
